lib/Mojo/Redis/Database.pm

```diff
@@ -121,14 +121,20 @@ sub _add_method {
   }
 }
 
+sub _connection {
+  my ($self, $non_blocking) = @_;
+  $self->connection($self->redis->_blocking_connection) if !$non_blocking and !$self->{connection};
```

Comment:
Unsure about this; isn't the connection attribute used for non-blocking connections? Would storing the blocking connection there break later non-blocking queries from the same $db?

Reply:
I need to store the blocking connection on the database object, so the next blocking call reuse the same connection.

I considered adding a check so you cannot go from blocking to non-blocking and the other way around. I don't think I want a single $db to be able to do both.

Reply:
How about if I add the check and die if the $db is used in both context, so don't get into a "deadlock" situation?